starknet_devnet/blueprints/rpc/misc.py

```diff
@@ -3,13 +3,64 @@
 """
 
 from __future__ import annotations
+from itertools import islice
+from queue import Empty
+from typing import Union, List
+import collections
+from starknet_devnet.blueprints.rpc.structures.types import (
+    BlockId,
+    Felt,
+    Address,
+)
+from starknet_devnet.blueprints.rpc.structures.responses import (
+    RpcEventsResult,
+)
 
-from typing import Union
-
-from starknet_devnet.blueprints.rpc.structures.payloads import Felt, Address
 from starknet_devnet.state import state
 
 
+def is_empty(value):
+    """
+    Return true if empty.
+    """
+    return not value and isinstance(value, collections.Iterable)
+
+
+def is_not_empty(value):
+    """
+    Return true if not empty.
+    """
+    return not is_empty(value)
+
+
+def filter_address(address, event):
+    """
+    Filter by address.
+    """
+    return bool(address == "" or event.from_address == int(address, 0))
+
+
+def filter_keys(keys, event):
+    """
+    Filter by keys.
+    """
+    return bool(
+        (is_empty(keys)) or (is_not_empty(keys) and bool(set(event.keys) & set(keys)))
+    )
+
+
+def get_events_from_block(block, address, keys):
+    """
+    Return filtered events.
+    """
+    events = []
+    for event in block.transaction_receipts[0].events:
```

Comment:
Note: implementation with one transaction per block

Reply:
I thought that currently we support only one transaction in block

Reply:
Changed for all events in all transaction_receipts